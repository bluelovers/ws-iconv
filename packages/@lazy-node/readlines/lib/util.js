"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.splitBufferByBuffer = exports.bufferIndexWith = exports.bufferIndexWithByBuffer = exports.bufferStripEndWith = exports.bufferStripEndWithByBuffer = exports.bufferEndWith = exports.bufferEndWithByBuffer = exports.handleOptionNewLineCharacter = void 0;
function handleOptionNewLineCharacter(newLineCharacter) {
    if (!newLineCharacter) {
        return [10 /* EnumNewLineCharacter.LF */];
    }
    else if (typeof newLineCharacter !== 'number') {
        let ls = [];
        for (let i = 0; i < newLineCharacter.length; i++) {
            let c = newLineCharacter[i];
            if (typeof c !== 'number') {
                ls.push(c.charCodeAt(0));
            }
            else {
                ls.push(c);
            }
        }
        return ls;
    }
    return [newLineCharacter];
}
exports.handleOptionNewLineCharacter = handleOptionNewLineCharacter;
function bufferEndWithByBuffer(buf, value, byteOffset, encoding) {
    let i = buf.lastIndexOf(value, byteOffset !== null && byteOffset !== void 0 ? byteOffset : (0 - value.length), encoding);
    return i !== -1 && (i + value.length) === buf.length;
}
exports.bufferEndWithByBuffer = bufferEndWithByBuffer;
function bufferEndWith(buf, value, byteOffset, encoding) {
    if (typeof value === 'number') {
        return buf[buf.length - 1] === value;
    }
    let i = buf.lastIndexOf(value, byteOffset !== null && byteOffset !== void 0 ? byteOffset : (0 - value.length), encoding);
    return i !== -1 && (i + value.length) === buf.length;
}
exports.bufferEndWith = bufferEndWith;
function bufferStripEndWithByBuffer(buf, value, byteOffset, encoding) {
    let i = buf.lastIndexOf(value, byteOffset !== null && byteOffset !== void 0 ? byteOffset : (0 - value.length), encoding);
    if (i !== -1 && (i + value.length) === buf.length) {
        return buf.slice(0, i);
    }
    return buf;
}
exports.bufferStripEndWithByBuffer = bufferStripEndWithByBuffer;
function bufferStripEndWith(buf, value, byteOffset, encoding) {
    if (typeof value === 'number') {
        if (buf[buf.length - 1] === value) {
            return buf.slice(0, buf.length - 1);
        }
    }
    else {
        let i = buf.lastIndexOf(value, byteOffset !== null && byteOffset !== void 0 ? byteOffset : (0 - value.length), encoding);
        if (i !== -1 && (i + value.length) === buf.length) {
            /*
            console.dir({
                i,
                vl: value.length,
                bl: buf.length,

                v: value.toString(),
                b: buf.toString(),
                b2: buf.slice(0, i).toString(),
            })
             */
            return buf.slice(0, i);
        }
    }
    return buf;
}
exports.bufferStripEndWith = bufferStripEndWith;
function bufferIndexWithByBuffer(buf, value, byteOffset, encoding) {
    let i = buf.indexOf(value, byteOffset, encoding);
    return i !== -1 && i === byteOffset;
}
exports.bufferIndexWithByBuffer = bufferIndexWithByBuffer;
function bufferIndexWith(buf, value, byteOffset, encoding) {
    if (typeof value === 'number') {
        return buf[byteOffset] === value;
    }
    let i = buf.indexOf(value, byteOffset, encoding);
    return i !== -1 && i === byteOffset;
}
exports.bufferIndexWith = bufferIndexWith;
function splitBufferByBuffer(buffer, value) {
    const lines = [];
    let bufferPosition = 0;
    let lastNewLineBufferPosition = 0;
    while (true) {
        let line;
        let bufferPositionValue = buffer[bufferPosition];
        if (bufferIndexWithByBuffer(buffer, value, bufferPosition++)) {
            line = buffer.slice(lastNewLineBufferPosition, bufferPosition += (value.length - 1));
            lines.push(line);
            lastNewLineBufferPosition = bufferPosition;
        }
        else if (bufferPositionValue === undefined) {
            break;
        }
    }
    let leftovers = buffer.slice(lastNewLineBufferPosition, bufferPosition);
    if (leftovers.length) {
        lines.push(leftovers);
    }
    return lines;
}
exports.splitBufferByBuffer = splitBufferByBuffer;
//# sourceMappingURL=util.js.map