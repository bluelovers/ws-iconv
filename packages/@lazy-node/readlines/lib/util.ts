import { INewLineCharacter, EnumNewLineCharacter } from './types';

export function handleOptionNewLineCharacter(newLineCharacter?: INewLineCharacter): number[]
{
	if (!newLineCharacter)
	{
		return [EnumNewLineCharacter.LF];
	}
	else if (typeof newLineCharacter !== 'number')
	{
		let ls = [] as number[]

		for (let i = 0; i < newLineCharacter.length; i++)
		{
			let c = newLineCharacter[i];

			if (typeof c !== 'number')
			{
				ls.push(c.charCodeAt(0))
			}
			else
			{
				ls.push(c)
			}
		}

		return ls
	}

	return [newLineCharacter]
}

export function bufferEndWithByBuffer(buf: Buffer,
	value: Uint8Array,
	byteOffset?: number,
	encoding?: BufferEncoding,
)
{
	let i = buf.lastIndexOf(value, byteOffset ?? (0 - value.length), encoding)

	return i !== -1 && (i + value.length) === buf.length
}

export function bufferEndWith(buf: Buffer,
	value: string | number | Uint8Array,
	byteOffset?: number,
	encoding?: BufferEncoding,
)
{
	if (typeof value === 'number')
	{
		return buf[buf.length - 1] === value
	}

	let i = buf.lastIndexOf(value, byteOffset ?? (0 - value.length), encoding)

	return i !== -1 && (i + value.length) === buf.length
}

export function bufferStripEndWithByBuffer(buf: Buffer,
	value: Uint8Array,
	byteOffset?: number,
	encoding?: BufferEncoding,
)
{
	let i = buf.lastIndexOf(value, byteOffset ?? (0 - value.length), encoding)

	if (i !== -1 && (i + value.length) === buf.length)
	{
		return buf.slice(0, i)
	}

	return buf
}

export function bufferStripEndWith(buf: Buffer,
	value: string | number | Uint8Array,
	byteOffset?: number,
	encoding?: BufferEncoding,
)
{
	if (typeof value === 'number')
	{
		if (buf[buf.length - 1] === value)
		{
			return buf.slice(0, buf.length - 1);
		}
	}
	else
	{
		let i = buf.lastIndexOf(value, byteOffset ?? (0 - value.length), encoding)

		if (i !== -1 && (i + value.length) === buf.length)
		{
			/*
			console.dir({
				i,
				vl: value.length,
				bl: buf.length,

				v: value.toString(),
				b: buf.toString(),
				b2: buf.slice(0, i).toString(),
			})
			 */

			return buf.slice(0, i)
		}
	}

	return buf
}

export function bufferIndexWithByBuffer(buf: Buffer,
	value: Uint8Array,
	byteOffset: number,
	encoding?: BufferEncoding,)
{
	let i = buf.indexOf(value, byteOffset, encoding)

	return i !== -1 && i === byteOffset
}

export function bufferIndexWith(buf: Buffer,
	value: string | number | Uint8Array,
	byteOffset: number,
	encoding?: BufferEncoding,)
{
	if (typeof value === 'number')
	{
		return buf[byteOffset] === value
	}

	let i = buf.indexOf(value, byteOffset, encoding)

	return i !== -1 && i === byteOffset
}

export function splitBufferByBuffer(buffer: Buffer, value: Uint8Array)
{
	const lines: Buffer[] = [];
	let bufferPosition = 0;

	let lastNewLineBufferPosition = 0;
	while (true)
	{
		let line: Buffer;
		let bufferPositionValue = buffer[bufferPosition];

		if (bufferIndexWithByBuffer(buffer, value, bufferPosition++))
		{
			line = buffer.slice(lastNewLineBufferPosition, bufferPosition += (value.length - 1));
			lines.push(line);
			lastNewLineBufferPosition = bufferPosition;
		}
		else if (bufferPositionValue === undefined)
		{
			break;
		}
	}

	let leftovers = buffer.slice(lastNewLineBufferPosition, bufferPosition);
	if (leftovers.length)
	{
		lines.push(leftovers);
	}

	return lines;
}
