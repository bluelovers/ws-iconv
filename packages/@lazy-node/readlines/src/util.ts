import { INewLineCharacter } from './types';
import { EnumNewLineCharacter } from './index';

export function handleOptionNewLineCharacter(newLineCharacter?: INewLineCharacter): number[]
{
	if (!newLineCharacter)
	{
		return [EnumNewLineCharacter.LF];
	}
	else if (typeof newLineCharacter !== 'number')
	{
		const ls = [] as number[];

		for (let i = 0; i < newLineCharacter.length; i++)
		{
			const c = newLineCharacter[i];

			if (typeof c !== 'number')
			{
				ls.push(c.charCodeAt(0))
			}
			else
			{
				ls.push(c)
			}
		}

		return ls
	}

	return [newLineCharacter]
}

