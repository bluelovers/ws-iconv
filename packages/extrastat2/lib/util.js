"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.reassemble = void 0;
const path_1 = require("path");
function reassemble(pathparts) {
    return (0, path_1.join)('/', ...pathparts);
}
exports.reassemble = reassemble;
//# sourceMappingURL=util.js.map