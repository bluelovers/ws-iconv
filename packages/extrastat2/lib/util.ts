import { join } from "path";

export function reassemble(pathparts: string[])
{
	return join('/', ...pathparts)
}
