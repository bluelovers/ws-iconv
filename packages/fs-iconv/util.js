"use strict";
/**
 * Created by user on 2019/3/17.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.trimFilename = void 0;
const sanitize_filename_1 = __importDefault(require("sanitize-filename"));
function trimFilename(name) {
    let ret = name.toString()
        .replace(/\r\n|\r|\n/g, ' ')
        .replace(/[\r\n\t  \xA0]+/g, ' ');
    return sanitize_filename_1.default(ret)
        .trim()
        .replace(/^[　\s_]+/g, '')
        .replace(/[　\s_]+$/g, '');
}
exports.trimFilename = trimFilename;
exports.default = exports;
//# sourceMappingURL=util.js.map