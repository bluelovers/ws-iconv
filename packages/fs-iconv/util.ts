/**
 * Created by user on 2019/3/17.
 */

import sanitize = require('sanitize-filename');

export function trimFilename(name): string
{
	let ret = name.toString()
		.replace(/\r\n|\r|\n/g, ' ')
		.replace(/[\r\n\t  \xA0]+/g, ' ')
	;

	return sanitize(ret)
		.trim()
		.replace(/^[　\s_]+/g, '')
		.replace(/[　\s_]+$/g, '')
		;
}

export default exports as typeof import('./util');
