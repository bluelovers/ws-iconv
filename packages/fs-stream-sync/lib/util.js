// @ts-ignore
Object.freeze(exports);
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoidXRpbC5qcyIsInNvdXJjZVJvb3QiOiIiLCJzb3VyY2VzIjpbInV0aWwudHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBQUEsYUFBYTtBQUNiLE1BQU0sQ0FBQyxNQUFNLENBQUMsT0FBTyxDQUFDLENBQUEiLCJzb3VyY2VzQ29udGVudCI6WyIvLyBAdHMtaWdub3JlXG5PYmplY3QuZnJlZXplKGV4cG9ydHMpXG4iXX0=