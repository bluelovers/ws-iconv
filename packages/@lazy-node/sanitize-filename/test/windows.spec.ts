import { inspect } from 'util';
import sanitizeFilename from '../index';

describe(`describe`, () =>
{

	[
		"con",
		"COM1",
		"PRN.",
		"aux.txt",
		"LPT9.asdfasdf",
		"LPT10.txt",
	].forEach(input =>
	{
		test(inspect(input), () =>
		{
			let actual = sanitizeFilename(input);

			expect(actual).toMatchSnapshot();

			console.log(inspect(input) , '=>', inspect(actual))
		})
	});

})
