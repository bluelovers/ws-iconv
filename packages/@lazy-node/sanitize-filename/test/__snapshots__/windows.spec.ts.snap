// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`describe 'COM1' 1`] = `"!"`;

exports[`describe 'LPT9.asdfasdf' 1`] = `"!.asdfasdf"`;

exports[`describe 'LPT10.txt' 1`] = `"!.txt"`;

exports[`describe 'PRN.' 1`] = `"!"`;

exports[`describe 'aux.txt' 1`] = `"!.txt"`;

exports[`describe 'con' 1`] = `"!"`;
